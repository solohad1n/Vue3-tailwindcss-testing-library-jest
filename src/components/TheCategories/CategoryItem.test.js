import { render, screen } from '@testing-library/vue'
import CategoryItem from './CategoryItem.vue'

const category = 'category'

function renderCategoryItem(isActive = false) {
  const options = {
    props: {
      category,
      isActive
    }
  }

  return render(CategoryItem, options)
}

it('renders with label', () => {

  renderCategoryItem()

  expect(screen.getByText(category)).not.toHaveClass('bg-gray-600 text-white')
})

it('renders active', () => {

  renderCategoryItem(true)

  expect(screen.getByText(category)).toHaveClass('bg-gray-600 text-white')
  
})

it('renders with title', () => {

  const { html } = renderCategoryItem()

  expect(html()).toMatchSnapshot()
})