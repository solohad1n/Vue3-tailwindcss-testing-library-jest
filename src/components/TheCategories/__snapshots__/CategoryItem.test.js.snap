// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"<a href=\\"#\\" class=\\"px-3 py-1 transition rounded-full border border-gray-300 bg-gray-100 hover:bg-gray-200\\">category</a>"`;

exports[`renders with label 1`] = `"<a href=\\"#\\" class=\\"px-3 py-1 transition rounded-full border border-gray-300 bg-gray-100 hover:bg-gray-200\\">category</a>"`;
